File: src/main/java/org/awesome/leetcode/algorithm/string/ZuoXuanZhuanZiFuChuanLcof.java
package org.awesome.leetcode.algorithm.string;

/**
 * 面试题58 - II. 左旋转字符串
 * @author jian
 */
public class ZuoXuanZhuanZiFuChuanLcof {
	
	public String reverseLeftWords(String s, int n) {
		return s.substring(n)+s.substring(0,n);
    }

}

File: leetcode/zuo-xuan-zhuan-zi-fu-chuan-lcof.go
package leetcode

func ReverseLeftWords(s string, n int) string {
	arr := []byte(s)
	Reverse1(arr, 0, len(s)-1)
	Reverse1(arr, 0, len(s)-n-1)
	Reverse1(arr, len(s)-n, len(s)-1)
	return string(arr)

}

func Reverse1(s []byte, start int, end int) {
	for start < end {
		s[start], s[end] = s[end], s[start]

		start++
		end--
	}
}
